octoprint_mrbeam/iobeam/laserhead_handler.py

```diff
@@ -0,0 +1,140 @@
+
+import os
+import yaml
+from octoprint_mrbeam.mrb_logger import mrb_logger
+
+# singleton
+_instance = None
+
+
+def laserheadHandler(plugin):
+	global _instance
+	if _instance is None:
+		_instance = LaserheadHandler(plugin)
+	return _instance
+
+
+class LaserheadHandler(object):
+	LASER_POWER_GOAL = 950
+
+	def __init__(self, plugin):
+		self._logger = mrb_logger("octoprint.plugins.mrbeam.iobeam.laserhead")
+		self._plugin = plugin
+		self._settings = plugin._settings
+		self._plugin_version = plugin._plugin_version
+
+		self._lh_cache = {}
+		self._last_used_lh_serial = None
+		self._correction_settings = {}
+		self._laser_heads_file = os.path.join(self._settings.getBaseFolder("base"), self._settings.get(['laser_heads', 'filename']))
+		self._load_laser_heads_file()  # Loads correction_settings, last_used_lh_serial and lh_cache
+
+		self._current_used_lh_serial = self._last_used_lh_serial
+
+	def set_current_used_lh_serial(self, serial):
+		if serial:
+			self._current_used_lh_serial = serial
+			self._write_laser_heads_file()
```

Comment:
write once all operations are done.

Reply:
I just noticed that I write the file here to save the last used laser head serial number. The thing is that if it is a known laser head, I would never write it otherwise. Maybe I could write it inside of an "else" after "if serial not in self._lh_cache"